src/LinuxTracing/TracerImpl.cpp

```diff
@@ -1371,11 +1371,19 @@ uint64_t TracerImpl::ProcessSampleEventAndReturnTimestamp(const perf_event_heade
     DeferEvent(event);
 
   } else if (is_sched_switch_with_callchain) {
-    // TODO(b/243510000): the implementation of this case will be added later
+    pid_t pid = ReadSampleRecordPid(ring_buffer);
```

Comment:
Can't this be -1 like mentioned in `PerfEventReaders.cpp`? Maybe it's fine to not have the callstack in those special cases, but let's be aware.

Reply:
This only happens if the thread is exiting. Thus the callstack would also only be the exit syscall. We are fine discarding. I added a comment.